### PR TITLE
Don't mege GSoC work on progress

### PR DESCRIPTION
**Pull request by rasanjanap**
Please don't merge this. GSoC work on progress

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/fhir/swagger/docs/License.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package org.openmrs.module.fhir.swagger.docs;
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/docs/License.java]: Please add licence header

File: omod/src/main/java/org/openmrs/module/fhir/swagger/docs/ExternalDocs.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.openmrs.module.fhir.swagger.docs;
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/docs/ExternalDocs.java]: Add licence header

File: omod/src/main/java/org/openmrs/module/fhir/swagger/docs/License.java (hunk 1 above)
Comment (rasanjanap) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/docs/License.java]: Corrected it

File: omod/src/main/java/org/openmrs/module/fhir/swagger/docs/ExternalDocs.java (hunk 2 above)
Comment (rasanjanap) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/docs/ExternalDocs.java]: Corrected it

File: omod/src/main/java/org/openmrs/module/fhir/filter/SwaggerForwardingFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		openmrsPath = fc.getServletContext().getContextPath();
+	}
+
+	@Override
+	public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
+		HttpServletRequest request = (HttpServletRequest) req;
+		String requestURI = request.getRequestURI();
+
+		if (requestURI.startsWith(openmrsPath + "/module/fhir/rest/swagger.json")) {
+			String newURI = requestURI.replace("/openmrs/module/fhir/rest/swagger.json", "/ms/fhir/fhirDocServelet");
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/filter/SwaggerForwardingFilter.java]: Let's move this strings constants to constant class

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationController.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            baseUrl.append(request.getServerName());
+            if ((scheme.equals("http") && port != 80) || (scheme.equals("https") && port != 443)) {
+                baseUrl.append(':');
+                baseUrl.append(request.getServerPort());
+            }
+
+            baseUrl.append(request.getContextPath());
+            String resourcesUrl = Context.getAdministrationService().getGlobalProperty(FHIRConstants.URI_PREFIX_GLOBAL_PROPERTY_NAME, baseUrl.toString());
+            String urlWithoutScheme = "";
+            String basePath = "/openmrs/ws/fhir";
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationController.java]: move to constants

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+import java.util.List;
+import java.util.Map;
+
+public class SwaggerSpecificationCreator {
+
+    private SwaggerSpecification swaggerSpecification;
+    private Conformance conformance;
+    private String baseUrl;
+    private String basePath;
+    Map<String, Definition> definitionMap = new HashMap<String, Definition>();
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: add private or public identifier

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 6]
```diff
@@ -0,0 +1,641 @@
...
+                            createOperationsMap = create.getOperations();
+                            if(createOperationsMap == null) {
+                                createOperationsMap = new HashMap<String, Operation>();
+                            }
+                        } else {
+                            create = new Path();
+                            createOperationsMap = new HashMap<String, Operation>();
+                        }
+                        Operation createOperation = new Operation();
+                        //createOperation.setDescritpion(SwaggerDocConstants.CREATE_RESOURCE + " " + resourceName + " "
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: remove commented fields

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 7]
```diff
@@ -0,0 +1,641 @@
...
+                        //                                + SwaggerDocConstants.RESOURCE_BY_ID);
+                        readOperation.setSummary(SwaggerDocConstants.RETURNS + " " + resourceName + " " + SwaggerDocConstants.DETAILS_OF_GIVEN_ID);
+                        List<String> produces = new ArrayList<String>();
+                        //Set mime type supported
+                        produces.add(SwaggerDocConstants.PRODUCES_JSON);
+                        produces.add(SwaggerDocConstants.PRODUCES_XML);
+                        List<CodeDt> formats = conformance.getFormat();
+                        for(CodeDt format : formats) {
+                            produces.add(format.getValue());
+                        }
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: generally add some comments

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 8]
```diff
@@ -0,0 +1,641 @@
...
+                            updateOperationsMap = update.getOperations();
+                            if(updateOperationsMap == null) {
+                                updateOperationsMap = new HashMap<String, Operation>();
+                            }
+                        } else {
+                            update = new Path();
+                            updateOperationsMap = new HashMap<String, Operation>();
+                        }
+                        Operation updateOperation = new Operation();
+                        //updateOperation.setDescritpion(SwaggerDocConstants.CREATE_RESOURCE + " " + resourceName + " "
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: remove commened fields

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 9]
```diff
@@ -0,0 +1,641 @@
...
+                            deleteOperationMap = delete.getOperations();
+                            if(deleteOperationMap == null) {
+                                deleteOperationMap = new HashMap<String, Operation>();
+                            }
+                        } else {
+                            delete = new Path();
+                            deleteOperationMap = new HashMap<String, Operation>();
+                        }
+                        Operation deleteOperation = new Operation();
+                        //deleteOperation.setDescritpion(SwaggerDocConstants.DELETE_DESCRIPTION + " " + resourceName + " "
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: this as well

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 10]
```diff
@@ -0,0 +1,641 @@
...
+                            searchOperationMap = search.getOperations();
+                            if(searchOperationMap == null) {
+                                searchOperationMap = new HashMap<String, Operation>();
+                            }
+                        } else {
+                            search = new Path();
+                            searchOperationMap = new HashMap<String, Operation>();
+                        }
+                        Operation searchOperation = new Operation();
+                        //searchOperation.setDescritpion(SwaggerDocConstants.SEARCH_RESOURCE + " " + resourceName + " " + SwaggerDocConstants.SEARCH_RESOURCE_BY_PARAMETERS);
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: same here

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 11]
```diff
@@ -0,0 +1,641 @@
...
+                        searchOperation.setParameters(parametersAvailable);
+                        searchOperationMap.put(SwaggerDocConstants.GET, searchOperation);
+                        search.setOperations(searchOperationMap);
+                        pathMap.put(pathId, search);
+                    }
+                }
+                createDefinition(resourceName);
+            }
+
+            //Add $everything
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: needed more description

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 12]
```diff
@@ -0,0 +1,641 @@
...
+    private String createSwaggerSpecification() {
+        String json = "";
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.configure(SerializationFeature.INDENT_OUTPUT, true);
+            mapper.configure(SerializationFeature.WRITE_NULL_MAP_VALUES, true);
+            mapper.setSerializationInclusion(Include.NON_NULL);
+            mapper.getSerializerProvider().setNullKeySerializer(new NullSerializer());
+            json = mapper.writeValueAsString(swaggerSpecification);
+        } catch (Exception exp) {
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: any reason for catching generic exception also remove print stack trace

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java [hunk 13]
```diff
@@ -0,0 +1,641 @@
...
+        parameter.setName(SwaggerDocConstants.FORMAT_PARAM);
+        parameter.setDescription(SwaggerDocConstants.FORMAT_PARAM_DESC);
+        parameter.setIn(SwaggerDocConstants.IN_QUERY);
+        parameter.setRequired(false);
+        parameters.put(SwaggerDocConstants.FORMAT_PARAM_NAME, parameter);
+        swaggerSpecification.setParameters(parameters);
+    }
+
+    private String getSchemaRef(String resourceName) {
+        return "#/definitions/" + resourceName;
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: move string contants to constant class

File: omod/src/main/java/org/openmrs/module/fhir/swagger/docs/NullSerializer.java [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+public class NullSerializer extends JsonSerializer<Object> {
+
+    @Override
+    public void serialize(Object arg0, JsonGenerator arg1, SerializerProvider arg2) throws IOException,
+            JsonProcessingException {
+        arg1.writeFieldName("");
+
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/docs/NullSerializer.java]: formatting issue

File: omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java (hunk 12 above)
Comment (rasanjanap) [inline omod/src/main/java/org/openmrs/module/fhir/swagger/SwaggerSpecificationCreator.java]: @harsha89 there is nothing to do by catching specific exception. I have remove the print stack trace and added log

**Pull request merged**